e2e/extensions-protocol.e2e.spec.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * This tests the extension protocol handler.
+ * An E2E test is required to have access to the web page context.
+ */
+
+import path from 'path';
+
+import { ElectronApplication, Page, test, expect } from '@playwright/test';
+
+import { NavPage } from './pages/nav-page';
+import {
+  createDefaultSettings, getResourceBinDir, retry, startRancherDesktop, teardown,
+} from './utils/TestUtils';
+
+import { ContainerEngine, Settings } from '@pkg/config/settings';
+import { spawnFile } from '@pkg/utils/childProcess';
+
+/** The top level source directory, assuming we're always from the tree */
+const srcDir = path.dirname(path.dirname(__filename));
+const rdctl = executable('rdctl');
+
+/**
+ * Get the given executable. Similar to @pkg/utils/resources, but does not use
+ * Electron.app (which doesn't work during the test).
+ */
+function executable(name: string) {
+  const exeName = name + (process.platform === 'win32' ? '.exe' : '');
+
+  return path.join(srcDir, 'resources', process.platform, 'bin', exeName);
+}
+
+test.describe.serial('Extensions protocol handler', () => {
+  let app: ElectronApplication;
+  let page: Page;
+  let isContainerd = false;
+
+  async function ctrctl(...args: string[]) {
+    let tool = executable('nerdctl');
+
+    if (!isContainerd) {
+      tool = executable('docker');
+      if (process.platform !== 'win32') {
+        args = ['--context', 'rancher-desktop'].concat(args);
+      }
+    }
+
+    return await spawnFile(tool, args,
+      {
+        stdio: 'pipe',
+        env:   {
+          ...process.env,
+          PATH: `${ process.env.PATH }${ path.delimiter }${ getResourceBinDir() }`,
+        },
+      });
+  }
+
+  test.beforeAll(async() => {
+    createDefaultSettings({ kubernetes: { enabled: false } });
+    app = await startRancherDesktop(__filename, { mock: false });
+    page = await app.firstWindow();
+  });
+
+  test.afterAll(() => teardown(app, __filename));
+
+  test('should load backend', async() => {
+    await (new NavPage(page)).progressBecomesReady();
+  });
+
+  test('determine container engine in use', async() => {
+    const { stdout } = await spawnFile(rdctl, ['list-settings'], { stdio: 'pipe' });
+    const settings: Settings = JSON.parse(stdout);
+
+    expect(settings.containerEngine.name).toMatch(/^(?:containerd|moby)$/);
+    isContainerd = settings.containerEngine.name === ContainerEngine.CONTAINERD;
+  });
+
+  test('wait for buildkit', async() => {
+    test.skip(!isContainerd, 'Not running containerd, no need to wait for buildkit');
+
```

Comment:
Could you add a comment explaining what `buildctl debug info` is supposed to do?